ts/session/apis/open_group_api/sogsv3/sogsV3SendReaction.ts

```diff
@@ -0,0 +1,91 @@
+import { AbortSignal } from 'abort-controller';
+import { Data } from '../../../../data/data';
+import { OpenGroupReactionResponse, Reaction } from '../../../../types/Reaction';
+import { getEmojiDataFromNative } from '../../../../util/emoji';
+import { OnionSending } from '../../../onions/onionSend';
+import { OpenGroupPollingUtils } from '../opengroupV2/OpenGroupPollingUtils';
+import { batchGlobalIsSuccess, parseBatchGlobalStatusCode } from './sogsV3BatchPoll';
+
+export const hasReactionSupport = async (id: number): Promise<boolean> => {
+  const found = await Data.getMessageByServerId(id);
+  if (!found) {
+    window.log.warn(`Open Group Message ${id} not found in db`);
+    return false;
+  }
+
+  const conversationModel = found?.getConversation();
+  if (!conversationModel) {
+    window.log.warn(`Conversation for ${id} not found in db`);
+    return false;
+  }
+
+  if (!conversationModel.hasReactions) {
+    window.log.warn("This open group doesn't have reaction support. Server Message Id", id);
+    return false;
+  }
+
+  return true;
+};
+
+export const sendSogsReactionOnionV4 = async (
+  serverUrl: string,
+  room: string,
+  abortSignal: AbortSignal,
+  reaction: Reaction,
+  blinded: boolean
+): Promise<boolean> => {
+  const allValidRoomInfos = OpenGroupPollingUtils.getAllValidRoomInfos(serverUrl, new Set([room]));
+  if (!allValidRoomInfos?.length) {
+    window?.log?.info('getSendReactionRequest: no valid roominfos got.');
+    throw new Error(`Could not find sogs pubkey of url:${serverUrl}`);
+  }
+
+  const canReact = await hasReactionSupport(reaction.id);
+  if (!canReact) {
+    return false;
+  }
+
+  // for an invalid reaction we use https://emojipedia.org/frame-with-an-x/ as a replacement since it cannot rendered as an emoji
+  const emoji = getEmojiDataFromNative(reaction.emoji) ? reaction.emoji : '🖾';
+  const endpoint = `/room/${room}/reaction/${reaction.id}/${emoji}`;
+  const method = reaction.action === 0 ? 'PUT' : 'DELETE';
```

Comment:
you've got a enum so let's use !
```
  const method = reaction.action === Action.REACT ? 'PUT' : 'DELETE';
```

Reply:
Very good catch. Still getting used to using enums